eZ/Publish/Core/Persistence/Legacy/Content/Type/Gateway.php

```diff
@@ -15,177 +15,139 @@
 use eZ\Publish\Core\Persistence\Legacy\Content\StorageFieldDefinition;
 
 /**
- * Base class for content type gateways.
+ * Content Type Gateway.
+ *
+ * @internal For internal use by Persistence Handlers.
  */
 abstract class Gateway
 {
+    public const CONTENT_TYPE_TO_GROUP_ASSIGNMENT_TABLE = 'ezcontentclass_classgroup';
+    public const CONTENT_TYPE_GROUP_TABLE = 'ezcontentclassgroup';
+    public const CONTENT_TYPE_TABLE = 'ezcontentclass';
+    public const CONTENT_TYPE_NAME_TABLE = 'ezcontentclass_name';
+    public const FIELD_DEFINITION_TABLE = 'ezcontentclass_attribute';
+    public const MULTILINGUAL_FIELD_DEFINITION_TABLE = 'ezcontentclass_attribute_ml';
+
+    public const CONTENT_TYPE_GROUP_SEQ = 'ezcontentclassgroup_id_seq';
+    public const CONTENT_TYPE_SEQ = 'ezcontentclass_id_seq';
+    public const FIELD_DEFINITION_SEQ = 'ezcontentclass_attribute_id_seq';
+
     /**
-     * Inserts the given $group.
-     *
-     * @param \eZ\Publish\SPI\Persistence\Content\Type\Group $group
+     * Insert the given $group.
```

Comment:
`function insertGroup(Group $group)` and other like `Update a group based on the given GroupUpdateStruct` or ` Return a number of Types in Group.` are not necessary.